test-source/test/ceylon/time/testDates.ceylon

```diff
@@ -182,59 +182,32 @@ shared test void testWithDay() {
     assertTrue( data_1982_12_13.withDay(17) <=> date( 1982, december, 17) == equal);
 }
 
-shared test void testWithDay40() {
-    try {
-        data_1982_12_13.withDay(40);
-        fail("Should throw exception...");
-    } catch( AssertionError e ) {
-        assertTrue(e.message.contains("Invalid date"));
-    }
-}
+void assertThrowsInvalidDate(void operation())
+        => assertThat(operation).hasMessage((String message) => message.contains("Invalid date value"));
 
-shared test void testWithDay0() {
-    try {
-        data_1982_12_13.withDay(0);
-        fail("Should throw exception...");
-    } catch( AssertionError e ) {
-        assertTrue(e.message.contains("Invalid date"));
-    }
-}
+shared test void testWithDay40()
+        => assertThrowsInvalidDate(() => data_1982_12_13.withDay(40));
```

Comment:
Maybe like this would read better?
```ceylon
shared test void testWithDay40() => assertThrowsInvalidDate {
    void operation() { data_1982_12_13.withDay(40); }
};
```